pkg/slo-controller/noderesource/noderesource_controller.go

```diff
@@ -137,7 +139,7 @@ func (r *NodeResourceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	handler := config.NewColocationHandlerForConfigMapEvent(r.Client, *sloconfig.NewDefaultColocationCfg(), r.Recorder)
 	r.cfgCache = handler
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Node{}).
+		For(&corev1.Node{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
The node-resource controller should watch the change in the `Node.Status`. e.g. The batch resource allocatable would be updated when the node allocatable changed.

Reply:
You are indeed correct!